### PR TITLE
[TD-544] Tyk Dashboard Helm Chart

### PR DESCRIPTION
**Pull request by abhishiekgowtham**
As planned, this PR splits the dashboard component into its own individual chart. I don't think smoke tests will pass for these changes, maybe it's better to disable them on  `refactor/chart-split` branch. In order to test the run `make test-dashboard` from the root of the project.

### CODE REVIEW

**Review by vesko-tyk — CHANGES_REQUESTED** [review]
@abhishiekgowtham I've reviewed your PR and have left some comments. Please let me know know your thoughts on it. Thanks!

File: charts/dashboard/docs/doc.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+project:
+  name: Tyk Dashboard
+  url: https://github.com/TykTechnologies/tyk-helm-chart
+  description: This chart deploys a Tyk Dashboard on a Kubernetes cluster using the Helm package manager.
+  app: Tyk dashboard automation for K8s
+repository:
+  url: https://cloudsmith.io/~tyk/repos/helm/packages/
+  name: dashboard
```
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/docs/doc.yaml]: Commit `7f4f891` should address it.

File: charts/dashboard/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,165 @@
+{{- if .Values.enabled -}}
```
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/templates/deployment.yaml:1]: This is to render the templates conditionally. It's enabled by default (set to `true`). When importing these charts into an umbrella chart, it gives the parent chart an option.

File: charts/dashboard/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
```
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/templates/deployment.yaml:40]: Have addressed this issue in `7f4f891`

File: charts/dashboard/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+          - name: TYK_DB_NODESECRET
+            valueFrom:
+              secretKeyRef:
+                name: {{ if .Values.global.secrets.useSecretName }} {{ .Values.global.secrets.useSecretName }} {{ else }} secrets-{{ include "tyk-pro.fullname" . | replace "-dashboard" "" }} {{ end }}
+                key: APISecret
+          - name: TYK_DB_LICENSEKEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ if .Values.global.secrets.useSecretName }} {{ .Values.global.secrets.useSecretName }} {{ else }} secrets-{{ include "tyk-pro.fullname" . | replace "-dashboard" "" }} {{ end }}
+                key: DashLicense
```
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/templates/deployment.yaml]: This should be resolved by commit `7f4f891`.

File: charts/dashboard/docs/doc.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/docs/doc.yaml]: Fixed in ba872ec

File: charts/dashboard/templates/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  name: {{ $fullName }}
+  labels:
+    {{- include "dashboard.labels" . | nindent 4 }}
+{{- if .Values.ingress.annotations }}
+  annotations:
+  {{- range $key, $value := .Values.ingress.annotations }}
+    {{ $key }}: {{ $value | quote }}
+  {{- end }}
+{{- end }}
+spec:
```
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/templates/ingress.yaml:20]: Fixed in ba872ec

**Review by alephnull — APPROVED** [review]
This looks for a first cut.

**Review by gwotg — APPROVED** [review]
Thanks @abhishiekgowtham! LGTM

**Comment by gwotg** [comment]
> Thanks @abhishiekgowtham! LGTM

Do we plan to get rid of those smoke tests errors before merging?

File: charts/dashboard/scripts/init.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e
+
+apk --no-cache add curl jq
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+
+while [[ $(kubectl get po -n {{ .Release.Namespace }} --selector=app.kubernetes.io/component=tyk-dashboard -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do sleep 10; done &&
```
[the remark this answers is not in the document]
Comment (abhishiekgowtham) [reply charts/dashboard/scripts/init.sh:9]: yes, it does since that label exists for all dashboard pods.

**Comment by abhishiekgowtham** [comment]
> > Thanks @abhishiekgowtham! LGTM
> 
> Do we plan to get rid of those smoke tests errors before merging?

No, that's a different issue and not part of this PR. It'll be tackled differently once all (parent and child) charts are in place.

**Review by ermirizio — APPROVED** [review]
Looks Good!

**Review by vesko-tyk — APPROVED** [review]
LGTM!

**Pull request merged**